deps: Bump eslint suite to v9

0b5vr opened the pull request:
This PR bumps eslint suite to v9.

This also migrates the existing eslint configs to the new flat config.

See the commit logs for details.

0b5vr committed:
deps: Bump eslint to v9 + migrate to flat config

Migrated the eslint config using `npx @eslint/migrate-config .eslintrc.json`, and did not work flawlessly, obviously

I had to include config to examples into the root config file.
Instead, we are able to delete `.eslintrc` that existed for each example directory though

`require-atomic-updates` (has been set to `off`) does not exist in the default configs, so removed the line

`@typescript-eslint/no-unnecessary-type-assertion` requires a type assertion, removed it for now

0b5vr committed:
refactor: fix a lint issue

0b5vr committed:
deps: remove eslint related deps that are no longer used

0b5vr committed:
Merge branch 'dev' into eslint-v9

yue4u commented on packages/three-vrm-animation/src/VRMAnimationLoaderPlugin.ts:
```diff
@@ -205,7 +205,9 @@ export class VRMAnimationLoaderPlugin implements GLTFLoaderPlugin {
           parentBoneName = VRMHumanBoneParentMap[parentBoneName];
         }
-        parentBoneName ?? (parentBoneName = 'hipsParent');
+        if (parentBoneName == null) {
+          parentBoneName = 'hipsParent';
+        }
```
Do you mean
```suggestion
        parentBoneName ??= 'hipsParent'
```

yue4u reviewed (commented):
Tested locally and found two issues
1. `.eslintrc.json` is a leftover?
2. lint runs fine for packages but shows warning when opening `eslint.config.mjs` and `jest.config.mjs`

0b5vr commented:
> 1. `.eslintrc.json` is leftover?

`.eslintrc.json` is definitely a leftover. I'm gonna remove it.

> 2. lint runs fine for packages but shows warning when opening `eslint.config.mjs` and `jest.config.mjs`

seem to be errors of `@typescript-eslint/naming-convention` that can be easily fixed. I will take care of this.

0b5vr committed:
chore: remove `.eslintrc.json` that replaced by `eslint.config.mjs`

Addresses: https://github.com/pixiv/three-vrm/pull/1565#pullrequestreview-2656296176

0b5vr committed:
refactor: fix eslint issues in `eslint.config.mjs` and `jest.config.mjs`

These `@typescript-eslint/naming-convention` errors can be easily fixed by inserting brackets

Addresses: https://github.com/pixiv/three-vrm/pull/1565#pullrequestreview-2656296176
